config/config-api/src/main/java/com/thoughtworks/go/config/materials/mercurial/HgMaterialConfig.java

```diff
@@ -170,5 +157,12 @@ public void setConfigAttributes(Object attributes) {
         if (map.containsKey(URL)) {
             this.url = new HgUrlArgument((String) map.get(URL));
         }
+        if (map.containsKey("userName")) {
```

Comment:
This can be moved to `ScmMaterialConfig` as this is now supported by all materials

Reply:
The case of for SVN material, the key is `username`, so left it as is. Also: this method is only used by the rails based clicky admin pages, so I did not feel it appropriate to do any major changes.